### PR TITLE
Rename Range to AbstractRange

### PR DESCRIPTION
**Pull request by nalimilan**
Fixes https://github.com/JuliaLang/julia/issues/21901.

### CODE REVIEW

**Comment by nalimilan** [comment]
OK to merge? This tends to accumulate conflicts quite fast.

File: base/abstractarray.jl [hunk 1]
```diff
@@ -172,7 +172,7 @@ first(a::AbstractArray) = a[first(eachindex(a))]
     first(coll)
 
 Get the first element of an iterable collection. Returns the start point of a
```
Comment (mbauman) [inline base/abstractarray.jl]: pedant: a -> an

File: doc/src/manual/control-flow.md [hunk 2]
```diff
@@ -409,7 +409,7 @@ julia> for i = 1:5
 5
 ```
 
-Here the `1:5` is a `Range` object, representing the sequence of numbers 1, 2, 3, 4, 5. The `for`
+Here the `1:5` is an `AbstractRange` object, representing the sequence of numbers 1, 2, 3, 4, 5. The `for`
```
Comment (mbauman) [inline doc/src/manual/control-flow.md]: It'd be clearer to just say it's a `UnitRange` or maybe even just "a range" (lowercase).

File: doc/src/manual/arrays.md [hunk 3]
```diff
@@ -348,7 +348,7 @@ indices and can be converted to such by [`to_indices`](@ref):
 2. An array of scalar indices. This includes:
     * Vectors and multidimensional arrays of integers
     * Empty arrays like `[]`, which select no elements
-    * `Range`s of the form `a:c` or `a:b:c`, which select contiguous or strided subsections from `a` to `c` (inclusive)
+    * `AbstractRange`s of the form `a:c` or `a:b:c`, which select contiguous or strided subsections from `a` to `c` (inclusive)
```
Comment (mbauman) [inline doc/src/manual/arrays.md]: I'd simply say: "Ranges like `a:c`…"  Making this more jargon-y implies a distinction that doesn't exist.

**Review by mbauman — APPROVED** [review]
Minor comments about some of the doc changes.  Bravo for digging into the editor config files. 👍

**Comment by nalimilan** [comment]
Thanks for the review. Indeed the systematic replacement of `Range` to `AbstractRange` made some sentences quite weird. Should be fixed now.

**Pull request merged**